cluster/gce/gci/configure-helper.sh

```diff
@@ -2076,6 +2076,45 @@ function start-fluentd-resource-update {
   wait-for-apiserver-and-update-fluentd &
 }
 
+# VolumeSnapshot CRDs are installed by cluster addon manager, which may
+# not be available at this point. Run this as a background process.
+function wait-for-apiserver-and-update-volumesnapshot {
+  # Wait until volumesnapshot CRDs is in place.
+  echo "Wait until volume snapshot CRDs are installed"
+  until kubectl get volumesnapshotclasses.snapshot.storage.k8s.io
+  do
+    sleep 10
+  done
+
+  until kubectl get volumesnapshotcontents.snapshot.storage.k8s.io
+  do
+    sleep 10
+  done
+
+  until kubectl get volumesnapshots.snapshot.storage.k8s.io
+  do
+    sleep 10
+  done
+
+  echo "Wait until volume snapshot RBAC rules are installed"
+  until kubectl get clusterrolebinding volume-snapshot-controller-role
+  do
+    sleep 10
+  done
+
+  echo "Wait until volume snapshot controller is installed"
+  until kubectl get statefulset volume-snapshot-controller | grep volume-snapshot-controller | grep "1/1"
+  do
+    sleep 10
+  done
+}
+
+# Trigger background process that will ultimately update volumesnapshotresource
+# requirements.
+function start-volumesnapshot-resource-update {
+  wait-for-apiserver-and-update-volumesnapshot &
```

Comment:
Looking at `wait-for-apiserver-and-update-fluentd` above, it seems to be waiting for some objects to be initialized before applying some modification.  I'm not sure we need to do the same for snapshots. Especially because this function is run in the background, I'm not really sure how useful it is.  We can probably remove it and not wait for anything.

Reply:
Sure, I'll give a try.

Reply:
Removed it but then everything is failing.  Nothing from cluster addon manager gets installed.  I don't even see logs from cluster addon.  So I have to add it back.